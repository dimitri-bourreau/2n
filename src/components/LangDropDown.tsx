import React, { FunctionComponent, ReactElement, useState } from 'react';
import { useTranslation } from 'react-i18next';

const LangDrowDown: FunctionComponent = (): ReactElement => {
  const { i18n } = useTranslation();
  const [langDropOpen, setlangDropOpen] = useState(false);

  const toggleLangDropdown = (): null => {
    setlangDropOpen(!langDropOpen);
    return null;
  };

  const handleChangeLanguage = (newLanguage: string): null => {
    i18n.changeLanguage(newLanguage);
    return null;
  };

  return (
    <div id="langDropDown">
      {/* <Dropdown
        open={langDropOpen}
        toggle={toggleLangDropdown}
        className="d-table"
        size="sm"
      >
        <DropdownToggle theme="white">
          <span role="img" aria-label="globle">
            🌐
          </span>
        </DropdownToggle>
        <DropdownMenu right>
          <DropdownItem onClick={(): null => handleChangeLanguage('en')}>
            English
          </DropdownItem>
          <DropdownItem onClick={(): null => handleChangeLanguage('fr')}>
            Français
          </DropdownItem>
        </DropdownMenu>
      </Dropdown> */}
    </div>
  );
};

export default LangDrowDown;
