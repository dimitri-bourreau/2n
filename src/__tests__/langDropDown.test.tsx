import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import LangDropDown from '../components/LangDropDown';

describe('<LangDropDown />', () => {
  test('LangDropDown component should match snapshot', () => {
    const tree = renderer.create(<LangDropDown />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('LangDropDown should render correctly', () => {
    const root = document.createElement('div');
    ReactDOM.render(<LangDropDown />, root);
    expect(root.querySelector('#langDropDown')).not.toBeNull();
    expect(
      root.querySelector('#langDropDown button.btn.btn-white'),
    ).not.toBeNull();
  });
});
