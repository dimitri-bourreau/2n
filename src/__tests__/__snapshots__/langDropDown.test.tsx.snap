// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LangDropDown /> LangDropDown component should match snapshot 1`] = `
<div
  id="langDropDown"
>
  <div
    className="d-table btn-group-sm dropdown"
  >
    <button
      aria-haspopup={true}
      className="btn btn-white"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-label="globle"
        role="img"
      >
        üåê
      </span>
    </button>
    <div
      className="dropdown-menu dropdown-menu-right"
      role="menu"
      tabIndex="-1"
    >
      <button
        className="dropdown-item"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        English
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        Fran√ßais
      </button>
    </div>
  </div>
</div>
`;
